outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
+    4. Templates should have a CSS class that matches the template name, and other Blaze suggestions
+    5. Meteor ES2015 style guide (link to it, don't reproduce all of it)
+    6. Perhaps this should include copypasta boilerplate - a standard template/component, a standard publication, standard route, etc? Basically the stuff a scaffolding framework would generate for you?
+5. Small app
+    1. Directory structure around features, not client/server
```

Comment:
I am saying that there are publish endpoints and methods which are linked to a feature. And there are some which are not. For example, publish blog posts can be used not just in display of blog posts, but also in stats, and in notifications, and in tagging view and so on. While update password method is probably used only from the update form view.

So you have two types of publish functions and methods. One connected to a view/feature. And one independent which are reused across many features. (Often they are pretty generic, like "all posts from the user".)